### PR TITLE
React installed

### PR DESCRIPTION
**Pull request by lucy2512**

### CODE REVIEW

File: weather-app/package.json [hunk 1]
```diff
@@ -0,0 +1,38 @@
+{
+  "name": "weather-app",
```
Comment (rajdeep-ghosh) [inline weather-app/package.json]: Change the name from "weather-app" to "weather-app-client"

File: weather-app/public/index.html [hunk 2]
```diff
@@ -0,0 +1,43 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
```
Comment (rajdeep-ghosh) [inline weather-app/public/index.html]: Remove this favicon link, we don't need a favicon right now (remove only the link tag)

File: weather-app/public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta
+      name="description"
+      content="Web site created using create-react-app"
+    />
```
Comment (rajdeep-ghosh) [inline weather-app/public/index.html]: Change the content of the description to something relevant

File: weather-app/public/index.html [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
```
Comment (rajdeep-ghosh) [inline weather-app/public/index.html]: Remove this line (only link tag)

File: weather-app/public/index.html [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      manifest.json provides metadata used when your web app is installed on a
+      user's mobile device or desktop. See https://developers.google.com/web/fundamentals/web-app-manifest/
+    -->
+    <link rel="manifest" href="%PUBLIC_URL%/manifest.json" />
```
Comment (rajdeep-ghosh) [inline weather-app/public/index.html:17]: Remove this line

File: weather-app/public/index.html [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    <!--
+      Notice the use of %PUBLIC_URL% in the tags above.
+      It will be replaced with the URL of the `public` folder during the build.
+      Only files inside the `public` folder can be referenced from the HTML.
+
+      Unlike "/favicon.ico" or "favicon.ico", "%PUBLIC_URL%/favicon.ico" will
+      work correctly both with client-side routing and a non-root public URL.
+      Learn how to configure a non-root public URL by running `npm run build`.
+    -->
+    <title>React App</title>
```
Comment (rajdeep-ghosh) [inline weather-app/public/index.html]: Change the title to something relevant

**Review by rajdeep-ghosh — CHANGES_REQUESTED** [review]
Consider this changes!

**Review by rajdeep-ghosh — CHANGES_REQUESTED** [review]
Also, change the React JS folder name from "weather-app" to "client"

File: weather-app/public/index.html (hunk 5 above)
Comment (lucy2512) [reply weather-app/public/index.html:17]: I think this line is not be a harmful line for our project.

**Comment by lucy2512** [comment]
And what about the other two json files which had been created by default the React??

**Review by lucy2512 — COMMENTED** [review]
All changes have been made.

**Review by lucy2512 — COMMENTED** [review]
Complete!

**Review by DEBADYUTI-KARMAKAR — DISMISSED** [review]

**Review by lucy2512 — COMMENTED** [review]
done

**Review by lucy2512 — COMMENTED** [review]
Completely done

File: client/src/app.js [hunk 7]
```diff
@@ -3,7 +3,9 @@ dotenv.config();
 import express from 'express';
 import bodyParser from 'body-parser';
 import ejs from "ejs";
+import React from 'react';
 import https from "https";
+import './App.css';
```
Comment (rajdeep-ghosh) [inline client/src/app.js]: Remove imports of React and CSS

File: client/src/app.js [hunk 8]
```diff
@@ -3,7 +3,9 @@ dotenv.config();
...
 
 const app = express();
```
Comment (rajdeep-ghosh) [inline client/src/app.js]: Create a new file named `app.js` in the root directory of the project and move all the code from `client/src/app.js` to that file. And in `client/src/app.js` just render a simple `Hello World` onto the viewport

**Review by rajdeep-ghosh — CHANGES_REQUESTED** [review]
Consider these changes

File: client/src/app.js (hunk 8 above)
Comment (rajdeep-ghosh) [reply client/src/app.js]: The content of the `app.js` file is server-side code and everything inside the `client` folder is frontend code. We should not put backend code in frontend and must separate any business logic from UI to maintain the separation of concerns

File: .env.example [hunk 9]
```diff
@@ -1 +1 @@
-API_KEY=YOUR_WEATHERDATA_API_KEY
+API_KEY=5e5d00005c097b2f3725a058ccf405d6
```
Comment (rajdeep-ghosh) [inline .env.example]: We don't need to modify the `.env.example` file, it is just a reference file that users can consult if they need clarification on setting up their API KEY. For the application to work correctly and fetch weather data we need to create a `.env` file in the root directory of the project and put the OpenWeatherMap API KEY there, as you did here. You can also refer to the README.md file. Please discard this change and restore the original data

File: client/src/app.js [hunk 10]
```diff
@@ -0,0 +1,4 @@
+ReactDOM.render(
+    <h1>Hello, world!</h1>,
+    document.getElementById('root')
+  );
```
Comment (rajdeep-ghosh) [inline client/src/app.js]: We don't need to render again as we are already doing once in the `index.js` file. All we have to do is create a functional component and return the `h1` and then export the App component.

For example,

```suggestion
import React from 'react';

function App() {
  return (
    <div>
      <h1>Hello, World!</h1>
    </div>
  );
}

export default App;
```

**Review by rajdeep-ghosh — CHANGES_REQUESTED** [review]
Please look into these changes

File: .env.example (hunk 9 above)
Comment (rajdeep-ghosh) [reply .env.example]: In the **.env.example** file consider these changes

```suggestion
API_KEY=YOUR_OPENWEATHERMAP_API_KEY
```

Comment (lucy2512) [reply .env.example]: I know that ,sorry that was my mistake.

File: client/src/app.js (hunk 10 above)
Comment (lucy2512) [reply client/src/app.js]: [https://github.com/Bong-Ninjas/weather-app/pull/9#discussion_r601352215](url)
You have told me earlier to render a hello world that's why i have done this.

**Comment by lucy2512** [comment]
Decide first what to do then request for changes,I think that should be better.

**Review by rajdeep-ghosh — APPROVED** [review]
LGTM!

**Comment by rajdeep-ghosh** [comment]
Thankyou for your hardwork!

**Pull request merged**